src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,88 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private static Map<Class<?>, Class<?>> implementations;
+    private static final Map<Class<?>, Object> instances = new HashMap<>();
+    private static final String NO_ANNOTATION
+            = "Could not find class of %s annotated with @Component";
+    private static final String CREATION_EXCEPTION = "Could not create instance of class ";
+    private static final String FIELD_INIT_EXCEPTION = "Can`t initialize field ";
+
+    private Injector() {
+        implementations = new HashMap<>();
+        implementations.put(FileReaderService.class, FileReaderServiceImpl.class);
+        implementations.put(ProductParser.class, ProductParserImpl.class);
+        implementations.put(ProductService.class, ProductServiceImpl.class);
```

Comment:
recomendation from CM:
"Make Interface Implementations map a class field. You can fill it in using Map.of()."